core/src/Yaya/Fold/Native.hs

```diff
@@ -47,33 +50,54 @@ import "this" Yaya.Pattern
     XNor (Both, Neither),
     uncurry,
   )
+import "this" Yaya.Strict (IsNotStrict, IsStrict, Strict)
 
 -- | A fixed-point constructor that uses Haskell's built-in recursion. This is
 --   strict/recursive.
 newtype Fix f = Fix {unFix :: f (Fix f)}
 
+type instance Strict Fix = 'True
+
+type instance Strict (Fix f) = Strict f
+
 instance Projectable (->) (Fix f) f where
   project = unFix
 
 instance Steppable (->) (Fix f) f where
   embed = Fix
 
-instance (Functor f) => Recursive (->) (Fix f) f where
+-- |
+--
+--  __TODO__: @`IsStrict` (`Fix` f)@ should be implied by @`IsStrict` f@.
+instance
+  (IsStrict f, IsStrict (Fix f), Functor f) =>
+  Recursive (->) (Fix f) f
+  where
   cata ɸ = ɸ . fmap (cata ɸ) . project
 
-instance (Functor f, Foldable f, Eq1 f) => Eq (Fix f) where
+-- | When the pattern functor is not `Strict`, `Fix` may be used corecursively.
+instance (IsNotStrict (Fix f), Functor f) => Corecursive (->) (Fix f) f where
+  ana ψ = embed . fmap (ana ψ) . ψ
```

Comment:
This is an interesting consequence – when the functor is non-strict, `Fix` can be used corecursively. However, the same can’t be done with `Mu`, because `embed` (and `project`) on `Mu` requires `cata`, which has the `IsStrict (Mu f)` constraint, which would conflict with the `IsNotStrict (Mu f)` constraint required by `ana`.